### DISCUSSION TITLE
integrations/vscode/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#15827] [Docs]
# integrations/vscode/

An overview of how the Ultralytics-Snippets extension for Visual Studio Code can help developers accelerate their work with the Ultralytics Python package.

https://docs.ultralytics.com/integrations/vscode/

**Comment by Hugofox07** [comment]
Hi, I'm new to YOLO, but I'd like to know how to program it directly from VS Code instead of Google Colab. Does anyone have any videos with simple examples of how to display an image, or even a video on YOLO detection? Thanks for the help.

**Reply by glenn-jocher** [reply]
Thanks for posting here—yes, you can run Ultralytics YOLO locally in VS Code; install the package, then use these minimal examples, and the VS Code Snippets extension lets you insert `ultra.example-yolo-predict` to get a runnable template (the extension page linked above has a short walkthrough video); for setup see the Install Ultralytics guide and for options see the Predict mode guide.

```bash
pip install ultralytics
```

```python
from ultralytics import YOLO

# Image
model = YOLO("yolo11n.pt", task="detect")
model.predict(source="path/to/image.jpg", show=True)

# Video
model.predict(source="path/to/video.mp4", show=True, save=True)
```

See [Install Ultralytics](https://docs.ultralytics.com/quickstart/) and the [Predict mode guide](https://docs.ultralytics.com/modes/predict/); drop any follow-ups here so the team can see them.

**Reply by Hugofox07** [reply]
Thanks very much , for your help. thanks dude.

**Reply by glenn-jocher** [reply]
Glad it helped—if you run into anything else while experimenting with YOLO in VS Code, just drop your questions here and we’ll take a look.
